src/main/java/appeng/core/localization/GuiText.java

```diff
@@ -47,9 +47,20 @@
 	// tunnel names
 	METunnel, ItemTunnel, RedstoneTunnel, EUTunnel, FluidTunnel, OCTunnel, LightTunnel, RFTunnel, PressureTunnel,
 
-	StoredSize, CopyMode, CopyModeDesc, PatternTerminal, CraftingPattern,
-
-	ProcessingPattern, Crafts, Creates, And, With, MolecularAssembler,
+	StoredSize, CopyMode, CopyModeDesc, PatternTerminal,
+
+	// Pattern tooltips
+	CraftingPattern,
+	ProcessingPattern,
+	Crafts,
+	Creates,
+	And,
+	With,
+	Substitute,
+	Yes,
+	No,
+
+	MolecularAssembler,
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00115)